### DISCUSSION TITLE
Should I use `take: 1` or `take: -1` to get the most recent object?

### DISCUSSION THREAD

**Discussion by ghost** [prisma/prisma#3801] [Q&A] [answered]
I'm using `findFirst` like this:

```typescript
const mostRecentValue = await db.value.findFirst({
    where: { id },
    orderBy: { time: 'desc' }
})
```

`findFirst` uses `findMany` with `take: 1` under the hood (#3676), but I'm confused if "first" means the first value found descending or ascending.

I'm wondering if my `orderBy` option is necessary at all, too.

Thanks in advance!

**Comment by pantharshit00** [comment, marked as answer]
`take: -1` will give you the last record. You can read the patch notes of release which introduced the new API for more details: https://github.com/prisma/prisma/releases/tag/2.0.0-beta.7

You can also read the documentation page: https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/pagination#overview

**Reply by ghost** [reply]
Is `take` kinda like saying `LIMIT` in SQL? So would `findFirst` together with `orderBy` `desc` give the same result of something like an `findLast` api?

**Reply by pantharshit00** [reply]
Yes, you can think of take as a `LIMIT` in SQL. `orderBy desc` will give you a `findLast` result
